File: app/src/main/java/com/fyp/sahayogapp/custom/CustomTextView.java
package com.fyp.sahayogapp.custom;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.Canvas;
import android.graphics.Typeface;
import android.util.AttributeSet;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.core.content.res.ResourcesCompat;

import com.fyp.sahayogapp.R;

import java.lang.reflect.Type;


public class CustomTextView extends androidx.appcompat.widget.AppCompatTextView {
    String customFont;
    int cf;

    public CustomTextView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        style(context, attrs);
    }

    public CustomTextView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        style(context, attrs);
    }

    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);


    }

    private void style(Context context, AttributeSet attrs) {

        TypedArray a = context.obtainStyledAttributes(attrs,
                R.styleable.CustomTextView,0,0);
        try {
            cf = a.getInt(R.styleable.CustomTextView_fontName, 0);
        } finally {
            a.recycle();
        }
        Typeface typeface = null;
        int fontName = 0;
        switch (cf) {
            case 1:
               typeface = ResourcesCompat.getFont(context, R.font.lato_bold);

                break;
            case 2:
                typeface = ResourcesCompat.getFont(context, R.font.lato_light);
                break;

            case 4:
                typeface = ResourcesCompat.getFont(context, R.font.lato_thin);
                break;
            case 0 :
            default:
                typeface = ResourcesCompat.getFont(context, R.font.lato_regular);
                break;
        }


        setTypeface(typeface);

    }
}

File: app/src/main/java/com/fyp/sahayogapp/custom/CustomTextView.kt
package com.fyp.sahayogapp.custom

import android.content.Context
import androidx.appcompat.widget.AppCompatTextView
import android.util.AttributeSet
import android.graphics.Canvas
import android.content.res.TypedArray
import com.fyp.sahayogapp.R
import android.graphics.Typeface
import androidx.core.content.res.ResourcesCompat

class CustomTextView : AppCompatTextView {
    var customFont: String? = null
    var cf = 0

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        style(context, attrs)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    ) {
        style(context, attrs)
    }

    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)
    }

    private fun style(context: Context, attrs: AttributeSet?) {
        val a = context.obtainStyledAttributes(
            attrs,
            R.styleable.CustomTextView, 0, 0
        )
        cf = try {
            a.getInt(R.styleable.CustomTextView_fontName, 0)
        } finally {
            a.recycle()
        }
        var typeface: Typeface? = null
        val fontName = 0
        typeface = when (cf) {
            1 -> ResourcesCompat.getFont(context, R.font.lato_bold)
            2 -> ResourcesCompat.getFont(context, R.font.lato_light)
            4 -> ResourcesCompat.getFont(context, R.font.lato_thin)
            0 -> ResourcesCompat.getFont(context, R.font.lato_regular)
            else -> ResourcesCompat.getFont(context, R.font.lato_regular)
        }
        setTypeface(typeface)
    }
}